[Zhou Junmin] iP

ZhouJunmin opened the pull request.

ZhouJunmin committed:
week2

ZhouJunmin committed:
Level 1. Echo

Add Echo function to the renamed chatbot Kyrene

ZhouJunmin committed:
Level 2. Add, List

Add basic add-task functon and list-tasks function to the chatbot

ZhouJunmin committed:
Level 3. Mark as Done

Add isDone feature to show the status of the task

ZhouJunmin committed:
Level-4: ToDo, Event, Deadline

Rewrote classes so that task class is more succinct.
Added todo, event, and deadline subclasses from task class to achieve different task types.
Handled some potential null bugs.

ZhouJunmin committed:
Level-4

Fixed some bugs regarding event class

Aak242 commented on src/main/java/Kyrene.java:
```diff
@@ -0,0 +1,129 @@
+    final static String GREETING = "    Hi, I am Kyrene, your private reminder assistant.\n"
+            + "    What can I do for you?\n";
+    final static String BYE = "    Bye! Wish to see you again soon!\n";
+    final static String ERROR_TASK_NOT_EXIST = "    Sorry! This task does not exist.\n";
```
Good job on utilizing final variables to store printed Strings that won't change so you can reuse them throughout your code!

Aak242 commented on src/main/java/Kyrene.java:
```diff
@@ -0,0 +1,129 @@
+        }
+        else{
+            tasks[taskCount] = new Task(sentence);
+        }
```
I like that you've used multiple methods to make your code more readable and organized!

Aak242 commented on src/main/java/Kyrene.java:
```diff
@@ -0,0 +1,129 @@
+        System.out.print("\n");
+    }
+
+    public static boolean converseKyrene(String sentence){
+        boolean isEnd = false;
+        printDivider();
+        if(sentence.equals("bye")){
+            isEnd = true;
+            exitKyrene();
+            return isEnd;
+        }
+        else if(sentence.equals("list")){
+            printTasks();
+            printDivider();
+        }
+        else if(sentence.startsWith("mark ")){
+            int taskNumber = Integer.parseInt(sentence.substring(5));
+            markTask(taskNumber);
+        }
+        else if(sentence.startsWith("unmark ")){
+            int taskNumber = Integer.parseInt(sentence.substring(7));
+            unmarkTask(taskNumber);
+        }
+        else{
+            addTask(sentence);
+        }
+        return isEnd;
+    }
```
The if-else blocks would better follow the Java Coding Standards if they were placed in the following way:

if (condition) {
    statements;
} else {
    statements;
}

Aak242 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+        super(null);
+    }
+
+    public Deadline(String line){
+        super(line);
+        taskType = "D";
+        int dividerIndex = line.indexOf("/");
+        if(dividerIndex == -1){
+            setTaskName(line);
+            setDeadline(null);
+            return;
+        }
+        String deadline = line.substring(dividerIndex + 4);
+        setDeadline(deadline);
+        taskName = line.substring(0, dividerIndex - 1);
+        setTaskName(taskName);
+    }
```
Good job on consistently making your indentation 4 spaces

Aak242 reviewed (commented):
Code looks clean and well organized.
Needs some minor adjustments to better align with coding standards

ZhouJunmin committed:
Improvement on coding standard and coding quality based on Level-4

add MAX_ARRAY_LENGTH and change the command recognise block from if-else to switch

ZhouJunmin committed:
Level-5

Add missing-task, invalid command, missing-time exceptions into the chatbot

ZhouJunmin committed:
Leve-5 Handle Exceptions

Add task-not-found exception for mark/unmark

ZhouJunmin committed:
Improvement on coding quality

ZhouJunmin committed:
Merge branch 'branch-Level-5'

ZhouJunmin committed:
Improvement on Coding Standard

ZhouJunmin committed:
A-Packages

ZhouJunmin committed:
Level-6 Delete

ZhouJunmin committed:
Level-7 Save

ZhouJunmin committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/kyrene/Kyrene.java

ZhouJunmin committed:
Level-7 Save

fixed bug with not being able to update done status correctly when loading the file at initialisation.

ZhouJunmin committed:
A-Jar

ZhouJunmin committed:
Make error message more complete and specific

ZhouJunmin committed:
Fixed some potential bugs related to IndexOutOfBound

ZhouJunmin committed:
A-MoreOOP

Refractor main function into Ui, Command, TaskList, Storage, and Parser. Add enum Commands for easier command processings.

ZhouJunmin committed:
Level-8 Date and Time

Make Kyrene able to recognize date and time, and tell the user to re-enter date and time if format is unrecognizable. Kyrene is now able to list out events on a given date and deadlines before a certain time, with at and due commands respectively.

ZhouJunmin committed:
Merge pull request #1 from ZhouJunmin/branch-Level-8

Level-8 Date and Time

ZhouJunmin committed:
Level 9 Find

Kyrene is now able to search tasks in the to-do list with keyword.
Fixed some exception errors during refactoring in the previous version.
Make commands case-insensitive to be more user-friendly.

ZhouJunmin committed:
Merge pull request #2 from ZhouJunmin/branch-Level-9

Level 9 Find

okkhoy commented on src/main/java/kyrene/Kyrene.java:
```diff
@@ -0,0 +1,49 @@
+import java.io.FileNotFoundException;
+
+public class Kyrene {
+    private final static String FILE_PATH = "./data/Kyrene.txt";
```
👍

okkhoy commented on src/main/java/kyrene/command/Command.java:
```diff
@@ -0,0 +1,244 @@
+        this.taskNumber = -1;
+        this.description = null;
+        this.isExit = (command == Commands.BYE) || (command == Commands.INVALID);
+    }
+    public Command(Commands command, int taskNumber) {
```
Need some vertical spacing

okkhoy commented on src/main/java/kyrene/command/Command.java:
```diff
@@ -0,0 +1,244 @@
+        Ui.showTasks(tasks);
+    }
+
+    private void handleMark(TaskList tasks, Storage storage, boolean isDone) throws KyreneTaskNotFoundException {
```
All non-trivial methods need header comments.

okkhoy commented on src/main/java/kyrene/command/Command.java:
```diff
@@ -0,0 +1,244 @@
+        Ui.showSuccessMarkingTask(taskNumber, isDone);
+    }
+
+    private void handleAdd(TaskList tasks, Storage storage) throws KyreneInvalidCommandException, KyreneMissingTaskException {
```
line too long. consider splitting to 2 lines

okkhoy commented on src/main/java/kyrene/command/Command.java:
```diff
@@ -0,0 +1,244 @@
+
+    private void handleAdd(TaskList tasks, Storage storage) throws KyreneInvalidCommandException, KyreneMissingTaskException {
+        assert description != null;
+        String[] words = description.split(" ");
+        String classType = words[0];
```
Consider SLAPping it out. (similar to week 6 lecture example)

okkhoy commented on src/main/java/kyrene/command/Command.java:
```diff
@@ -0,0 +1,244 @@
+        }
+    }
+
+    private void handleAT(TaskList tasks) {
+        LocalDate date = Parser.parseDate(description);
+        TaskList events = new TaskList();
+        for (Task task : tasks) {
+            if (task.getTaskType().equals("E") && ((Event) task).isAt(date)){
+                events.add(task);
+            }
+        }
+        Ui.showEvents(events);
+    }
+
+    private void handleDue(TaskList tasks) {
+        LocalDateTime time = Parser.parseTime(description);
+        TaskList deadlines = new TaskList();
+        for (Task task : tasks) {
+            if (task.getTaskType().equals("D") && ((Deadline) task).isBefore(time)){
+                deadlines.add(task);
+            }
+        }
+        Ui.showDeadlines(deadlines);
+    }
+
+    private void handleFind(TaskList tasks) {
+        TaskList matches = new TaskList();
+        for (Task task : tasks) {
+            if (task.getTaskName().contains(description)) {
+                matches.add(task);
+            }
+        }
+        Ui.showMatches(matches);
+    }
```
These are good for modularity. However, consider refactoring them to other classes.

okkhoy commented on src/main/java/kyrene/parser/Parser.java:
```diff
@@ -0,0 +1,89 @@
+
+public class Parser {
+
+    public static Command parse(String sentence) throws NumberFormatException, ArrayIndexOutOfBoundsException, StringIndexOutOfBoundsException {
```
👍  good parse method

okkhoy commented on src/main/java/kyrene/task/Deadline.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.Scanner;
+
+public class Deadline extends Task{
+    protected static DateTimeFormatter formatterInput = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```
The pattern should be documented explicitly in your UG

okkhoy commented on src/main/java/kyrene/task/Event.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+    @Override
+    public String format() { return String.format("%b event %s /from %s /to %s\n", isDone, taskName, startTime.format(formatterInput), endTime.format(formatterInput));}
```
too long line

okkhoy reviewed (commented):
I hope these review comments help in polishing the iP code before submission.

ZhouJunmin committed:
A-JavaDoc

Add header comments to all non-private classes/methods, and non-trivial private methods.

ZhouJunmin committed:
Merge pull request #3 from ZhouJunmin/branch-A-JavaDoc

A-JavaDoc

ZhouJunmin committed:
A-UserGuide

Add a complete user guide of Kyrene V1.0 to README.md

ZhouJunmin committed:
Update README.md in docs folder

ZhouJunmin committed:
Edited docs/README.md for ip page

ZhouJunmin committed:
Edited docs/README.md
